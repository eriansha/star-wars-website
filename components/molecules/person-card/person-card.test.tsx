import { render } from '@testing-library/react'

import PersonCard from './person-card'

const props = {
  personId: 'asd',
  name: 'title'
}

describe('Person Card', () => {
  it('renders without crashing', () => {
    render(<PersonCard {...props} />)
  })

  it('render title props', () => {
    const { getByTestId } = render(<PersonCard {...props} />)
    expect(getByTestId('person-name')).toHaveTextContent(props.name)
  })

  it('contains href in anchor', () => {
    const { getByTestId } = render(<PersonCard {...props} />)
    expect(getByTestId('person-link')).toHaveAttribute(
      'href',
      `/people/${props.personId}`
    )
  })

  it('matches snapshot', () => {
    const tree = render(<PersonCard {...props} />)
    expect(tree.container).toMatchSnapshot()
  })
})
