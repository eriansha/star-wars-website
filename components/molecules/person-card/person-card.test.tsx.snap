// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person Card matches snapshot 1`] = `
<div>
  <div
    class="bg-white my-3 shadow-xl"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 200px; height: 200px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="film-poster"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <a
      data-testid="person-link"
      href="/people/asd"
    >
      <h5
        class="text-black"
        data-testid="person-name"
      >
        title
      </h5>
    </a>
  </div>
</div>
`;
